lib/containers/urls.pm

```diff
@@ -51,64 +51,220 @@ sub get_opensuse_registry_prefix {
+            totest   => sub {
+                'registry.suse.de/suse/sle-15-sp3/update/cr/totest/images/suse/sle15:15.3';
+            },
+            available_arch => ['x86_64', 'aarch64', 'ppc64le', 's390x']
+        },
+        '15-SP4' => {
+            released => sub { '' },
+            totest   => sub {
+                'registry.suse.de/suse/sle-15-sp4/update/cr/totest/images/suse/sle15:15.4 ';
+            },
+            available_arch => ['x86_64', 'aarch64', 'ppc64le', 's390x']
+        }
+    },
+    opensuse => {
+        Tumbleweed => {
+            released => sub { 'registry.opensuse.org/opensuse/tumbleweed' },
+            totest   => sub {
+                'registry.opensuse.org/' . get_opensuse_registry_prefix . 'opensuse/tumbleweed';
+            },
+            available_arch => ['x86_64', 'aarch64', 'ppc64le', 's390x', 'arm']
+        },
+        '15.0' => {
+            released => sub { 'registry.opensuse.org/opensuse/leap:15.0' },
+            totest   => sub {
+                'registry.opensuse.org/opensuse/leap/15.0/images/totest/images/opensuse/leap:15.0';
+            },
+            available_arch => ['x86_64']
+        },
+        '15.1' => {
+            released => sub { 'registry.opensuse.org/opensuse/leap:15.1' },
+            totest   => sub {
+                my $arch = shift;
+                if ($arch eq 'x86_64') {
+                    'registry.opensuse.org/opensuse/leap/15.1/images/totest/containers/opensuse/leap:15.1';
+                } elsif ($arch eq 'arm') {
+                    'registry.opensuse.org/opensuse/leap/15.1/arm/images/totest/containers/opensuse/leap:15.1';
+                } else {
+                    '';
+                }
+
+            },
+            available_arch => ['x86_64', 'arm']
+        },
+        '15.2' => {
+            released => sub { 'registry.opensuse.org/opensuse/leap:15.2' },
+            totest   => sub {
+                my $arch = shift;
+                if ($arch eq 'x86_64') {
+                    'registry.opensuse.org/opensuse/leap/15.2/images/totest/containers/opensuse/leap:15.2';
+                } elsif ($arch eq 'arm') {
+                    'registry.opensuse.org/opensuse/leap/15.2/arm/images/totest/containers/opensuse/leap:15.2';
+                } else {
+                    '';
+                }
+            },
+            available_arch => ['x86_64', 'arm']
+        },
+        '15.3' => {
+            released => sub { 'registry.opensuse.org/opensuse/leap:15.3' },
+            totest   => sub {
+                my $arch = shift;
+                if (grep { $_ eq $arch } qw/x86_64 aarch64 ppc64le s390x/) {
+                    'registry.opensuse.org/opensuse/leap/15.3/images/totest/containers/opensuse/leap:15.3';
+                } elsif ($arch eq 'arm') {
+                    'registry.opensuse.org/opensuse/leap/15.3/arm/images/totest/containers/opensuse/leap:15.3';
+                } else {
+                    '';
+                }
```

Comment:
Why do you return `''` instead of just skipping?

Reply:
is it a big issue? i didnt overthink about it to be honest. it can be neglected.

Reply:
I think it does, imagine some new image will arrive for testing, someone will schedule it but forget to add it here.
The tests may still pass and nobody will see that the image is actually not tested. I would skip the `else` block.

Then at the end you can check if some image has been returned and if not we can do something like:
```perl
die("No image for DISTRI=$distr VERSION=$version ARCH=$arch available");
```

Reply:
I agree with Pavel here. Better to use `die` here to not give a false green (the test won't test any image).

Reply:
I think this code will never be hitten, as we first call `check_image_arch_in_registry()` and thus, if someone give a arch which isn't there we already `die("Unknown combination of distro/arch.");`.

Reply:
This is also true. The condition will be `if('')` which will be evaluated to `false` and it will hit the `die`. So, up to you how to implement it.

Reply:
Exactly. If this won't be hitten I'd just remove the `else {}` completely.

Reply:
*this change is ready for review*

Reply:
There are still occasions like `released => sub { }` or `released => sub { }` and I'd remove those as well.

Reply:
those are fine and they have their purpose. Avoid `Can't use an undefined value`. simplifies invocation and error handling